### PR TITLE
fixes #6540, #8335 - Add error to vmware when no datacenter, save passwo...

### PR DESCRIPTION
**Pull request by shlomizadok**
Add error to vmware when no datacenter, save password when testing connection

### CODE REVIEW

File: app/views/compute_resources/form/_vmware.html.erb [hunk 1]
```diff
@@ -2,8 +2,8 @@
 <%= text_f f, :user %>
 <%= password_f f, :password, :placeholder => password_placeholder(f.object) %>
 <% datacenters = list_datacenters f.object%>
-<%= selectable_f(f, :uuid, datacenters, {}, {:label => _('Datacenter'),
-                 :help_inline => link_to_function(datacenters.empty? ? "Load Datacenters" : "Test Connection", "testConnection(this)",
+<%= selectable_f(f, :datacenter, datacenters, {}, {:label => _('Datacenter'),
```
Comment (dLobatog) [inline app/views/compute_resources/form/_vmware.html.erb:5]: `:uuid` is the attribute in the db and the attribute in the model, why changing it? It's shared across compute resources.

Comment (shlomizadok) [inline app/views/compute_resources/form/_vmware.html.erb:5]: Originally, changed that cause `:uuid` didn't have validation and datacenter did.
Anyhow, reverted and added `:uuid` to presence validation.

File: app/views/compute_resources/form/_vmware.html.erb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 <%= password_f f, :password, :placeholder => password_placeholder(f.object) %>
 <% datacenters = list_datacenters f.object%>
 <%= selectable_f(f, :uuid, datacenters, {}, {:label => _('Datacenter'),
-                 :help_inline => link_to_function(datacenters.empty? ? "Load Datacenters" : "Test Connection", "testConnection(this)",
+                 :help_block => link_to_function(datacenters.empty? ? "Load Datacenters" : "Test Connection", "testConnection(this)",
```
Comment (shlomizadok) [inline app/views/compute_resources/form/_vmware.html.erb:6]: helping @domcleal to [test] from within line note

**Comment by shlomizadok** [comment]
line note didn't trigger [test]

File: app/views/compute_resources/form/_vmware.html.erb (hunk 2 above)
Comment (domcleal) [inline app/views/compute_resources/form/_vmware.html.erb:6]: Oh sorry, I disabled it again, it wasn't working very well :)

**Comment by unorthodoxgeek** [comment]
@ohadlevy  -this is good to be merged.

File: app/models/compute_resources/foreman/model/vmware.rb [hunk 3]
```diff
@@ -265,8 +265,9 @@ def vm_hw_versions
 
     def test_connection(options = {})
       super
-      if errors[:server].empty? and errors[:user].empty? and errors[:password].empty?
+      if errors[:server].empty? && errors[:user].empty? && errors[:password].empty?
         update_public_key options
+        errors.delete(:datacenter) unless datacenters.empty?
```
Comment (ohadlevy) [inline app/models/compute_resources/foreman/model/vmware.rb]: this makes little sense?

File: app/models/compute_resources/foreman/model/vmware.rb [hunk 4]
```diff
@@ -6,7 +6,7 @@ class Vmware < ComputeResource
 
     include ComputeResourceConsoleCommon
 
-    validates :user, :password, :server, :datacenter, :presence => true
+    validates :user, :password, :server, :datacenter, :uuid, :presence => true
```
Comment (ohadlevy) [inline app/models/compute_resources/foreman/model/vmware.rb]: datacenter and uuid are the same db column, no need to run validation twice.

**Comment by ohadlevy** [comment]
merged as ebac69c thanks @shlomizadok

**Pull request closed without merging**